spec/integration/counters_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+require 'spec/support/integration'
+require 'spec/support/persistence'
+require 'spec/support/counters'
+require 'routemaster'
+
+describe 'Process counters', slow:true do
+  let(:processes) { Acceptance::ProcessLibrary.new }
+  before { WebMock.disable! }
+
+  before { ENV['ROUTEMASTER_COUNTER_FLUSH_INTERVAL'] = '1' }
+
+  shared_examples 'start and stop' do |count, type_tag|
+    before { subject.start }
+    after  { subject.terminate }
+
+    it 'starts cleanly' do
+      subject.wait_start
+      sleep 2 # enough time for the flusher thread to trigger
+      expect(get_counter('process', type_tag.merge(status: 'start'))).to eq(count)
+      # FIXME: use wait_log to avoid the sleeping (proper integration)
```

Comment:
Should we add a ticket for this rather than just a FIXME? As a general rule I ban TODO/FIXME from code and raise tickets instead.